File: AutoMap/gateway/app/models/map.py
from pydantic import BaseModel
from typing import Optional, List


class Location(BaseModel):
    lat: float
    lng: float


class Place(BaseModel):
    id: str
    name: str
    address: str
    location: Location
    category: str
    rating: float
    description: str
    image: str
    distance: float
    duration: Optional[str] = None


class RecommendRequest(BaseModel):
    query: str
    location: Optional[Location] = None


class RecommendResponse(BaseModel):
    places: List[Place]
    centerLocation: Location


class LocationResponse(BaseModel):
    lat: float
    lng: float
    address: str

File: frontend/src/types/map.ts
export interface Location {
  lat: number
  lng: number
}

export interface Place {
  id: string
  name: string
  address: string
  location: Location
  category: string
  rating: number
  description: string
  image: string
  distance: number
  duration?: string
}

export interface RecommendRequest {
  query: string
  location?: Location
}

export interface RecommendResponse {
  places: Place[]
  centerLocation: Location
}
